packages/projects-docs/pages/learn/integrations/tailscale.mdx

```diff
@@ -0,0 +1,116 @@
+</Callout>
+
+## Motivation
+
+Many applications exist as part of a collection of services.
+While things like mocks can help isolate a single application during development, this isn't always possible.
+Just because you develop in CodeSandbox, however, doesn't mean you can't connect your application to private services running in your organization's network.
+
+Integrating with Tailscale provides containers running in a CodeSandbox Repository environment with access to your private resources.
+Combined with Tailscale's Magic DNS and ACLs, you can develop your application as if it were hosted in the same network as your other services — without compromising security.
+
+## Creating an Auth Key
+
+Integrating a CodeSandbox Repository with Tailscale requires an [auth key](https://tailscale.com/kb/1085/auth-keys/), a secret you embed in the CodeSandbox environment to allow registration on your network.
+Network admins can create them on the [auth key page](https://login.tailscale.com/admin/settings/keys).
+
+The auth key you give to CodeSandbox should be:
+
+* **Reusable**, so you don't encounter issues when forking a new branch,
+* **Ephemeral**, so you don't have to clean up old branches from your tailnet, and
+* **Pre-authorized**, so you don't have to wait for someone to approve a new branch before you start developing.
+
+Depending on your tailnet setup, you may also want it to be:
+
+* **Tagged**, so CodeSandbox environments are automatically labeled and access-controlled via Tailscale's ACLs.
+
+## Setting up your Repository
+
+Tailscale currently works in CodeSandbox Repository environments by adding a container to a Dockerized application setup (for example, using `docker-compose`).
+The additional container runs the Tailscale client and provides networking for the other containers.
+
+### Environment Variables
+
+Before launching the new container, it is necessary to add the Tailscale auth key as an environment variable.
+With a branch open in the Repository editor, go to the CodeSandbox menu, and choose **Env variables**.
+Throughout the rest of this documentation, we will assume you named the variable `TS_AUTH_KEY`.
+Restart any existing branches to update the running environment.
+
+### Docker Container
+
+In a setup using Docker Compose, we can add the `tailscale` image as an additional service beside our application.
+Here's an example `docker-compose.yml` file with the Tailscale service:
+
+```yaml
+services:
+  tailscale:
+    image: tailscale/tailscale
+    cap_add:
+      - net_admin
+      - net_raw
+      - sys_module
+    command: "tailscaled --state=mem:"
+    volumes:
+      - "/dev/net/tun:/dev/net/tun"
+
+  myapp:
+    # ...
+    network_mode: service:tailscale
+```
+
+This runs the latest `tailscale/tailscale` image with a few customizations.
```

Comment:
Here you call it `tailscale/tailscale` but earlier you just call it `tailscale`. Pick one.